src/v/raft/heartbeat_manager.cc

```diff
@@ -115,10 +115,8 @@ heartbeat_manager::requests_for_range() {
                 .group = r->group(),
               };
               const auto raft_metadata = r->meta();
-              if (
-                _enable_lw_heartbeat()
-                && !needs_full_heartbeat(
-                  follower_metadata, raft_metadata, r->flushed_offset())) {
+              if (should_lw_heartbeat(
+                    r, follower_metadata, raft_metadata, r->flushed_offset())) {
```

Comment:
is it fair to say that this isn't a real memory leak, but rather a buffer with uncontrolled growth?